server/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java

```diff
@@ -917,7 +917,8 @@ protected boolean docValuesByDefault() {
     @Override
     protected void doXContentBody(XContentBuilder builder, boolean includeDefaults, Params params) throws IOException {
         super.doXContentBody(builder, includeDefaults, params);
-        if (includeDefaults || (mappedFieldType.isSearchable() && fieldType.indexOptions() != Defaults.FIELD_TYPE.indexOptions())) {
+        if (fieldType.indexOptions() != IndexOptions.NONE
```

Comment:
so we don't want to print out index_options: none ? sorry I don't recall what we were doing before the refactoring.

Reply:
Yes, if `index` is set to `false` then we don't emit the `index_options`.  The pre-existing method that converts the lucene `IndexOptions` enum to a string doesn' accept `IndexOptions.NONE`

Reply:
thanks for explaining, makes sense